### PR TITLE
Added null check for name in UserInfo class

### LINKED ISSUE #2897: userInfo inside pub login command doesn't contain user's name
## Description

I am new here and looking to contribute. Therefore, I stumbled upon the `pub login` command which uses oauth2 to authenticate the user and store their email & name into userInfo. But I tried with two different accounts and both resulted in a `null` for the name field.

Here is the response coming from `userinfo_endpoint`:

```dart
{
sub: 103068153864256870934, 
picture: https://lh3.googleusercontent.com/a-/AOh14GgynQMyqolmPXQ87dTgmpG5PQaLk75IlNlzEf-qUA=s96-c, 
email: tusharojha2001@gmail.com,
email_verified: true
}
``` 
Either we can remove the name property from the `UserInfo` class or the API should respond with the user's name.

### PR DESCRIPTION
**Pull request by tusharojha**
## Description

pub login command which uses oauth2 to authenticate the user and store their email & name into userInfo sometimes returns null for the name.

Fixes #2897

### CODE REVIEW

**Comment by tusharojha** [comment]
@sigurdm I have resolved some linter issues in the last commit. PTAL.

File: lib/src/ignore.dart [hunk 1]
```diff
@@ -532,7 +532,7 @@ class _IgnorePrefixPair {
   @override
   String toString() {
     // TODO: implement toString
```
Comment (sigurdm) [inline lib/src/ignore.dart:534]: Let's delete this stray todo
```suggestion
```

File: lib/src/command/login.dart [hunk 2]
```diff
@@ -60,5 +60,5 @@ class _UserInfo {
   final String email;
   _UserInfo(this.name, this.email);
   @override
-  String toString() => '<$email> "$name"';
+  String toString() => '<$email> ${name ?? ''}';
```
Comment (sigurdm) [inline lib/src/command/login.dart:63]: I don't like the extra space we get when name is null. Perhaps:

```suggestion
  String toString() => ['<$email>', if (name != null) name].join(' ');
```

**Review by sigurdm — APPROVED** [review]
Looks fine - only we should fix the extra space.

File: lib/src/ignore.dart (hunk 1 above)
Comment (tusharojha) [reply lib/src/ignore.dart]: Done.

**Comment by sigurdm** [comment]
Thanks a lot for the contribution!

**Pull request merged**